src/app/features/wallets/transfer/transfer.page.ts

```diff
@@ -0,0 +1,194 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute } from '@angular/router';
+import { NavController, Platform } from '@ionic/angular';
+import { TranslocoService } from '@ngneat/transloco';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { BehaviorSubject, combineLatest, of } from 'rxjs';
+import { catchError, concatMap, first, tap } from 'rxjs/operators';
+import { BlockingActionService } from '../../../shared/blocking-action/blocking-action.service';
+import { DiaBackendStoreService } from '../../../shared/dia-backend/store/dia-backend-store.service';
+import { DiaBackendWalletService } from '../../../shared/dia-backend/wallet/dia-backend-wallet.service';
+import { ErrorService } from '../../../shared/error/error.service';
+import { TransferLoadingComponent } from './transfer-loading/transfer-loading.component';
+import { TransferRequestSentComponent } from './transfer-request-sent/transfer-request-sent.component';
+
+@UntilDestroy({ checkProperties: true })
+@Component({
+  selector: 'app-transfer',
+  templateUrl: './transfer.page.html',
+  styleUrls: ['./transfer.page.scss'],
+})
+export class TransferPage {
+  mode = '';
+
+  readonly assetWalletBscNumBalance$ =
+    this.diaBackendWalletService.assetWalletBscNumBalance$;
+  readonly integrityWalletBscNumBalance$ =
+    this.diaBackendWalletService.integrityWalletBscNumBalance$;
+
+  transferAmount: number | null = null;
+
+  readonly gasFee$ = new BehaviorSubject<number>(0);
+  readonly totalCost$ = new BehaviorSubject<number>(0);
+
+  readyToSend = false;
+
+  keyboardIsShown = false;
+
+  private orderId = '';
+
+  constructor(
+    private readonly activeRoute: ActivatedRoute,
+    private readonly diaBackendWalletService: DiaBackendWalletService,
+    private readonly blockingActionService: BlockingActionService,
+    private readonly storeService: DiaBackendStoreService,
+    private readonly dialog: MatDialog,
+    private readonly navCtrl: NavController,
+    private readonly errorService: ErrorService,
+    private readonly translocoService: TranslocoService,
+    private readonly platform: Platform
+  ) {
+    this.activeRoute.paramMap.subscribe(paramMap => {
+      if (!paramMap.has('mode')) {
```

Comment:
Not really. Agree that in hybrid app users are not able to visit arbitrary route. Just add a check to be safe. Should I remove it?

Reply:
I think it's best to remove it for now, reasons are

1. It only works when something got very wrong, but when something got very wrong
    - If user frequently triggers the condition, we need to fix that wrong route anyway
    - If user seldom triggers the condition, letting them restart the app and everything go back to normal is an acceptable bottomline and then we still need to fix that wrong route anyway, so the need for this safeguard is not high, we can polish it a bit before implement it, see 2 ->

 2. If we want a generally safer usage of route and paramMap, we should consider adding redirection in `app-routing.module.ts` and adding [Route Guard](https://angular.io/api/router/CanActivate) because if we want to do it we should do it for all components that might have the same problem

Reply:
Ok! Will do so in another commit.